common/lib/ucdc_edx_api_plugin/ucdc_edx_api/api/views.py

```diff
@@ -0,0 +1,40 @@
+import logging
+
+from rest_framework import status
+from rest_framework.response import Response
+from rest_framework.views import APIView
+
+from openedx.core.lib.api.authentication import OAuth2AuthenticationAllowInactiveUser
+from openedx.core.lib.api.permissions import ApiKeyHeaderPermission
+from student.models import UserProfile
+from third_party_auth.models import OAuth2ProviderConfig
+from ucdc_edx_api.api.permissions import AnonPermissionOnly
+from ucdc_edx_api.api.serializers import (
+    UserRegisterSerializer,
+    UserSocialAuthSerializer,
+)
+
+log = logging.getLogger(__name__)
+
+
+class RegisterAPIView(APIView):
+    authentication_classes = (OAuth2AuthenticationAllowInactiveUser,)
+    permission_classes = (ApiKeyHeaderPermission, AnonPermissionOnly)
+
+    def post(self, request, *args, **kwargs):
+        params = request.data.copy()
+        user_serializer = UserRegisterSerializer(data=params)
+        if user_serializer.is_valid(raise_exception=True):
+            user = user_serializer.save()
+        UserProfile.objects.create(user=user, allow_certificate=True)
+        params["provider"] = OAuth2ProviderConfig.objects.first().backend_name
```

Comment:
Let's add a comment that `provider` should be only one or first in the admin dashboard.

Reply:
done